src/contraction/src/pgr_contract.hpp

```diff
@@ -141,38 +145,62 @@ Pgr_contract< G >::contract_to_level(G &graph,int64_t level)
 		remove_1_degree_vertices(graph);
 		break;
 		case 1:
-		//remove_2_degree_vertices(graph);
+		remove_2_degree_vertices(graph);
 		default:
 		// do nothing
 		;
 	}
 }
 
+
+
+template < class G >
+void
+Pgr_contract< G >::degreeMap(G &graph,std::ostringstream& dmap)
+{
+	//cout << "Printing degree_V map" << endl;
+      degree_to_V_i it1;
+      dmap << "This is a degree map " ;
+      Q_i it2;
+      for ( it1=degree_to_V_map.begin(); it1!=degree_to_V_map.end(); ++it1)
+      {
+        dmap << it1->first << "-->" ;
+        for (it2=it1->second.begin(); it2 !=it1->second.end(); ++it2)
+        {
+           dmap << graph.graph[(*it2)].id << ", ";
+        }
+        dmap << "\n";
+      }
+}
+
 template < class G >
 void
 Pgr_contract< G >::remove_1_degree_vertices(G &graph) {
 	EO_i out,out_end;
-	V front=degree_to_V_map[1].front();
-	int64_t frontid=-1;
-	frontid=graph.graph[front].id;
+	V front;
+	//errors << "first vertex: " << front ;
 	std::vector<V> one_degree_vertices=degree_to_V_map[1];
 	degree_to_V_i it;
-	//cout << "1 degree vertices " << one_degree_vertices.size()  << endl;
+	std::cout << "1 degree vertices " << one_degree_vertices.size();
+	//std::cout << "Front id" << frontid;
+	graph.m_num_vertices--;	
 	while (degree_to_V_map[1].size()>0)
 	{
-		//cout << "Front " << graph[front].id << endl;
+		//std::cout << "Front " << graph.graph[front].id;
+		V front=degree_to_V_map[1].front();
+		int64_t frontid=graph.graph[front].id;	
 		for (boost::tie(out, out_end) = out_edges(front, graph.graph);
 			out != out_end; ++out) 
 		{
 			V s=source(*out, graph.graph);
 			V t=target(*out, graph.graph);
-			auto source_id=graph.graph[s].id;
+			int64_t source_id=graph.graph[s].id;
```

Comment:
whoa... if the fix was becuase of this then the problem is other
we will play a socrates game later